### PR TITLE
1. PopoverAPI w/ redux (First)

### PR DESCRIPTION
**Pull request by mannycolon**
#### This pull request addresses:

Implement redux for the PopoverApi. #884 

#### How to test this pull request:

In order to test this PR you must pull the following branch from the tools (both scripturePane and the tool you're using to test) repo:
**feature/mc-showPopover**

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/890)

### CODE REVIEW

**Comment by klappy** [comment]
*[src/js/actions/CoreActionConsts.js, line 54 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoBuKoret2_vb1k7cB:-KeoBuKp-Cgpz8OOrNaY:b-twif80) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/actions/CoreActionConsts.js#L54)):*
> ```JavaScript
>   SELECT_MODAL_SECTION: "SELECT_MODAL_SECTION",
>   SHOW_POPOVER: "SHOW_POPOVER",
>   CLOSE_POPOVER: "CLOSE_POPOVER",
> ```

Looks like a trailing "," was overlooked.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by klappy** [comment]
*[src/js/actions/ReportsActions.js, line 2 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoD7qAOqqNBLfq0iBB:-KeoD7qAOqqNBLfq0iBC:b2ittrm) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/9207bda2350b3b7698a9a6cc30e9d293c897aa01/src/js/actions/ReportsActions.js#L2)):*
> ```JavaScript
> const consts = require('./CoreActionConsts');
> const api = window.ModuleApi;
> ```

I'm assuming we've removed the need for the ModuleApi now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by klappy** [comment]
*[src/js/actions/SideBarActions.js, line 2 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoDaxGuSGSyugVLB9r:-KeoDaxGuSGSyugVLB9s:b5wz8h0) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/actions/SideBarActions.js#L2)):*
> ```JavaScript
> const CheckStoreActions = require('./CheckStoreActions.js');
> const consts = require('./CoreActionConsts');
> ```

Could this be mounted in CheckStoreActions so that this could look like:
`const consts = CheckStoreActions.consts`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by klappy** [comment]
*[src/js/components/core/Popover.js, line 2 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoDz62Bw2FgicmG3I0:-KeoDz62Bw2FgicmG3I1:b9aeo0w) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/components/core/Popover.js#L2)):*
> ```JavaScript
> import React, { Component } from 'react'
> import { Popover } from 'react-bootstrap'
> ```

I'm loving this so much I'm wondering if we should make 'tc-tools-bootstrap' library so we can do something like:
`import { ScripturePane, VerseCheck, CheckInfo } from 'tc-tools-bootstrap'`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by klappy** [comment]
*[src/js/containers/PopoverContainer.js, line 15 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoF4cY_wlV9V6PEtC0:-KeoF4cY_wlV9V6PEtC1:b-36spp0) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/containers/PopoverContainer.js#L15)):*
> ```JavaScript
>   }
> }
> ```

Below this point is new to me. It looks good, but seems interesting to keep separate from the component. Out of curiosity, why is it not in there?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by klappy** [comment]
*[src/js/containers/PopoverContainer.js, line 28 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoFJvGLoegyXkIff7e:-KeoFJvGLoegyXkIff7f:b-m8pm2s) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/containers/PopoverContainer.js#L28)):*
> ```JavaScript
> }
> 
> module.exports = connect(
> ```

The connect is also new to me, I'm assuming it's a Redux convention?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by klappy** [comment]
*[src/js/reducers/coreStoreReducer.js, line 9 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoGT1StOZ8jUqUkIZI:-KeoGT1StOZ8jUqUkIZJ:bkz6ivs) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/9207bda2350b3b7698a9a6cc30e9d293c897aa01/src/js/reducers/coreStoreReducer.js#L9)):*
> ```JavaScript
> var PACKAGE_COMPILE_LOCATION = pathex.join(PARENT, 'packages-compiled')
> const merge = require('lodash.merge');
> const PACKAGE_SUBMODULE_LOCATION = pathex.join(window.__base, 'tC_apps');
> ```

Where is this being handled now?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by klappy** [comment]
Reviewed 24 of 28 files at r1.
Review status: 24 of 28 files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by EllDoubleYew** [comment]
Reviewed 28 of 28 files at r1.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[src/js/actions/ReportsActions.js, line 2 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoD7qAOqqNBLfq0iBB:-KeoI_REPfnyOLYXr4Rb:b-h4ke6x) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/9207bda2350b3b7698a9a6cc30e9d293c897aa01/src/js/actions/ReportsActions.js#L2)):*
<details><summary><i>Previously, klappy (Christopher Klapp) wrote…</i></summary><blockquote>

I'm assuming we've removed the need for the ModuleApi now.
</blockquote></details>

You are correct, In a perfect world we never reference window.ModuleApi ever again :sweat_smile: 

---

*[src/js/components/core/Popover.js, line 21 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoIyk9zz6RXuHrZKqp:-KeoIyk9zz6RXuHrZKqq:b-83yw2w) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/components/core/Popover.js#L21)):*
> ```JavaScript
>           arrowOffsetTop={-1000}
>           title={<span>
>                   {title}
> ```

Could we possibly build this outside of the return statement to De-anonymize this component? Its not a must-do thing but it would be helpful to better understand what exactly this is.

---

*[src/js/containers/PopoverContainer.js, line 28 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoFJvGLoegyXkIff7e:-KeoJjGErVRjnWzRQI47:b8y8q9w) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/containers/PopoverContainer.js#L28)):*
<details><summary><i>Previously, klappy (Christopher Klapp) wrote…</i></summary><blockquote>

The connect is also new to me, I'm assuming it's a Redux convention?
</blockquote></details>

Essentially what this is doing is creating a highly optimized container for us. Check out http://redux.js.org/docs/basics/UsageWithReact.html#presentational-and-container-components for more info @klappy

---

*[src/js/reducers/popoverReducer.js, line 31 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoKUmAwsG0MWKMvoVt:-KeoKUmAwsG0MWKMvoVu:bsh728e) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/reducers/popoverReducer.js#L31)):*
> ```JavaScript
>       return state;
>   }
> }
> ```

This reducer is beautiful :+1:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*

**Comment by EllDoubleYew** [comment]
First of all hats off to @mannycolon for this PR. This is really the nitty gritty stuff when it comes to a redux refactor. Things are really starting to shape up in terms of our data architecture. 

I am ready to merge this but would like to wait on a review from @RoyalSix as well because admittedly he is more versed in this domain than I. Good stuff manny!

**Comment by mannycolon** [comment]
Review status: 26 of 34 files reviewed at latest revision, 9 unresolved discussions.

---

*[src/js/actions/CoreActionConsts.js, line 54 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoBuKoret2_vb1k7cB:-KeoNYv-UeduW0Q_etfv:b82ygnm) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/actions/CoreActionConsts.js#L54)):*
<details><summary><i>Previously, klappy (Christopher Klapp) wrote…</i></summary><blockquote>

Looks like a trailing "," was overlooked.
</blockquote></details>

LOL I left it there just to make it easier for the net person that needs to add an action. so that they don't have to keep in mind that they need to add the coma lol

---

*[src/js/actions/ReportsActions.js, line 2 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoD7qAOqqNBLfq0iBB:-KeoNoGqgCrn9DtqyECo:b-u3mgsl) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/9207bda2350b3b7698a9a6cc30e9d293c897aa01/src/js/actions/ReportsActions.js#L2)):*
<details><summary><i>Previously, EllDoubleYew (Luke Wilson) wrote…</i></summary><blockquote>

You are correct, In a perfect world we never reference window.ModuleApi ever again :sweat_smile: 
</blockquote></details>

we are working towards achieving that 

---

*[src/js/actions/SideBarActions.js, line 2 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoDaxGuSGSyugVLB9r:-KeoNt1rvMjLMRLVQpNh:b9s6fm9) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/actions/SideBarActions.js#L2)):*
<details><summary><i>Previously, klappy (Christopher Klapp) wrote…</i></summary><blockquote>

Could this be mounted in CheckStoreActions so that this could look like:
`const consts = CheckStoreActions.consts`
</blockquote></details>

Unfortunately, we can't because the `consts` is not only used with `CheckStoreActions`

---

*[src/js/components/core/Popover.js, line 2 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoDz62Bw2FgicmG3I0:-KeoO8TSmBo9r_necLqQ:b-shtsw6) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/components/core/Popover.js#L2)):*
<details><summary><i>Previously, klappy (Christopher Klapp) wrote…</i></summary><blockquote>

I'm loving this so much I'm wondering if we should make 'tc-tools-bootstrap' library so we can do something like:
`import { ScripturePane, VerseCheck, CheckInfo } from 'tc-tools-bootstrap'`
</blockquote></details>

oh yeah we should definitely  do that. that's  a great idea!

---

*[src/js/components/core/Popover.js, line 21 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoIyk9zz6RXuHrZKqp:-KeoONE3F-99cHS2FKAI:be8u3ov) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/components/core/Popover.js#L21)):*
<details><summary><i>Previously, EllDoubleYew (Luke Wilson) wrote…</i></summary><blockquote>

Could we possibly build this outside of the return statement to De-anonymize this component? Its not a must-do thing but it would be helpful to better understand what exactly this is.
</blockquote></details>

yeah, that makes sense. I will do that. 

---

*[src/js/containers/PopoverContainer.js, line 15 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoF4cY_wlV9V6PEtC0:-KeoPjkG6oKGBS94oX_O:bt0uvpt) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/cd86912cc1a965781361d3f26e29186e0758b709/src/js/containers/PopoverContainer.js#L15)):*
<details><summary><i>Previously, klappy (Christopher Klapp) wrote…</i></summary><blockquote>

Below this point is new to me. It looks good, but seems interesting to keep separate from the component. Out of curiosity, why is it not in there?
</blockquote></details>

basically redux at its best! lol we should speak about this in a show and tell maybe.

---

*[src/js/reducers/coreStoreReducer.js, line 9 at r1](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890#-KeoGT1StOZ8jUqUkIZI:-KeoPywiMF8lAn2MpKXp:b32vmt) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/9207bda2350b3b7698a9a6cc30e9d293c897aa01/src/js/reducers/coreStoreReducer.js#L9)):*
<details><summary><i>Previously, klappy (Christopher Klapp) wrote…</i></summary><blockquote>

Where is this being handled now?
</blockquote></details>

??

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/unfoldingword-dev/translationcore/890)*
